commit 9b54cb1ec617ea8bff86d6d74c06cef5a51e26c9
Author: Patrik Martinsson <Patrik.Martinsson@smhi.se>
Date:   Thu Aug 26 17:39:44 2010 -0500

    dispatcher: add dhcp4-change and dhcp6-change events (bgo #563654)
    
    New dispatcher events on DHCP lease changes.

diff --git a/man/NetworkManager.8.in b/man/NetworkManager.8.in
index a6be073..4f700e9 100644
--- a/man/NetworkManager.8.in
+++ b/man/NetworkManager.8.in
@@ -52,6 +52,12 @@ A VPN connection has been deactivated.
 .TP
 .I "hostname"
 The system hostname has been updated.  Use gethostname(2) to retrieve it.
+.TP
+.I "dhcp4\-change"
+The DHCPv4 lease has changed (renewed, rebound, etc).
+.TP
+.I "dhcp6\-change"
+The DHCPv6 lease has changed (renewed, rebound, etc).
 .SH OPTIONS
 The following options are supported:
 .TP
diff --git a/src/nm-device.c b/src/nm-device.c
index d7c57fc..9e8ccdf 100644
--- a/src/nm-device.c
+++ b/src/nm-device.c
@@ -1344,6 +1344,7 @@ handle_dhcp_lease_change (NMDevice *device, gboolean ipv6)
 			nm_dhcp_client_foreach_option (priv->dhcp6_client,
 			                               dhcp6_add_option_cb,
 			                               priv->dhcp6_config);
+			nm_utils_call_dispatcher ("dhcp6-change", connection, device, NULL);
 		} else {
 			nm_log_warn (LOGD_DHCP6, "(%s): failed to update IPv6 config in response to DHCP event.",
 			             nm_device_get_ip_iface (device));
@@ -1368,6 +1369,7 @@ handle_dhcp_lease_change (NMDevice *device, gboolean ipv6)
 			nm_dhcp_client_foreach_option (priv->dhcp4_client,
 			                               dhcp4_add_option_cb,
 			                               priv->dhcp4_config);
+			nm_utils_call_dispatcher ("dhcp4-change", connection, device, NULL);
 		} else {
 			nm_log_warn (LOGD_DHCP6, "(%s): failed to update IPv4 config in response to DHCP event.",
 			             nm_device_get_ip_iface (device));
